great_expectations/expectations/core/expect_query_results_to_match_source.py

```diff
@@ -192,5 +207,70 @@ def _validate(
                 "result": {
                     "unexpected_count": unexpected_count,
                     "unexpected_percent": unexpected_percent,
+                    "details": {
+                        "missing_rows": missing_rows,
+                        "unexpected_rows": unexpected_rows,
+                    },
                 },
             }
+
+    @classmethod
+    def _compute_row_data(
+        cls,
+        col_names: list[str],
+        frequency_map: Counter[tuple],
+    ) -> list[dict[str, Any]]:
+        """Given a frequency map of values and a list of their keys, compute a list of
+        column-name -> column value dictionaries.
+
+        Example:
+          - col_names: ["a", "b"]
+          - frequency_map: Counter({
+                (1, 2): 2,
+                (3, 4): 1,
+            })
+
+          -> [
+                {"a": 1, "b": 2},
+                {"a": 1, "b": 2},
+                {"a": 3, "b": 4},
+             ]
+        """
+        row_values = sorted(frequency_map.elements(), key=cmp_to_key(cls._null_safe_tuple_compare))
```

Comment:
Can you add a comment about how this line works? I had to google multiple things to understand it.

Reply:
Updated with a bit of commenting, though trying to focus more comments on the "why" than the "what"